### DISCUSSION TITLE
Catch's new touch mode

### DISCUSSION THREAD

**Discussion by Jan-Arbus** [ppy/osu#20237] [Ideas] [answered]
I noticed that the current control acceleration is a bit painful
My idea is this: the control method is similar to the keyboard, the left controls the speed, and the right controls the direction
You can add a setting to toggle control mode
![屏幕截图 2022-09-11 183002](https://user-images.githubusercontent.com/74862110/189522828-1de1bc0b-0d92-41df-9a55-5039ac873036.png)

**Comment by msmcculley** [comment]
I second this. It seems really jarring from someone who plays osu!catch on desktop where one hand controls the dash speed and the other hand controls the movement of the catcher. Dashing in the current way is not reliable (accidentally hitting the "slow" direction instead of the hyper/dash direction), and thus is just frustrating and nonintuitive, but I am jaded in the sense that I have played osu!catch a certain way for over 5 years. It is nigh impossible to replicate a good/passing score after nearly any platter, let alone rain+ difficulty.

I think it would be beneficial to allow the player to also select which side of the screen should be the dash/controls because there are people who prefer to play the game a certain way.

**Comment by Glinte** [comment]
Catch player (rank ~1200) with 10d of playtime here. I can FC 5* maps normally but I cant even pass a 2* map with touch input. This input method is too different that my previous skills cannot be transferred except for my pattern recognition. To be fair though, you can definitely get used to this input method, but the width of the input bars should be at least 1.5x larger because I cant even consistently hit the bar on an ipad.

**Reply by peppy** [reply]
Definitely agree it should be wider. I thought this would be the case but I didn't test on an actual device until the latest release.

I chose to go with the proposed style because I think it will be pretty natural to new players, but as mentioned in touch related discussions, we'll be adding further customisation in the future to adjust the position and size of the individual buttons.

**Comment by Glinte** [comment]
Spent 1-2 hours playing with it this afternoon. Here are a few scattered thoughts about it.
### Possible Improvements
1. Wider bars
2. Move the 'hold for menu' button elsewhere
I keep accidentally touching it, during normal gameplay and also especially when transitioning into a break because the button moves up in height. Moving the button to bottom center seems like a natural choice.
### Bugs (on Android 12, Samsung Galaxy S21 Ultra)
not that interested in spending extra time to write a bug report so just documenting. These bugs does not happen on ipad.
1. When holding both left and right, sometimes the inputs are dropped, causing the catcher to jitter (Pretty reproducible)
2. Sometimes the dash button stay held for no reason (Happened twice out of ~10 plays, both times on my left dash)
### How I Play
maybe there are some insights to be gained here?
I use 4 fingers to control. When I want to dash, say, to the right, I would click both of the right buttons. Basically, I will never click the dash button without clicking the normal button. This seems to be slightly closer to the usual way of playing since now the dash button virtually only has the function of enabling dashing and no movement is caused by the dash button. I also keep my fingers to the top half of the playfield because the 'hold for menu' button is annoying.

**Comment by honguyenminh** [comment, marked as answer]
So, here's my 2 cents on this.

The current input method is fine for new players, but horizontal bars doesn't really make sense. Even if we have a wide aspect ratio, it's still way too cramped, fingers will slide around accidentally, and besides users would only use either the lower part or the higher part, leaving half of the space useless. I suggest make the default arrangement to be upper half = dashed, lower half = normal.This also opens up a new playstyle, like swiping up to dash and swipe down to go back to normal.

Look something like this
```
-----------------------------------------------------------------------
|                |                                   |                | 
|    Left dash   |                                   |   Right dash   | 
|                |                                   |                |
|----------------|             Play area             |----------------|
|                |                                   |                | 
|   Left normal  |                                   |  Right normal  | 
|                |                                   |                | 
-----------------------------------------------------------------------
```

**Reply by peppy** [reply]
I tend to agree that this is a better default layout. Can be implemented immediately if anyone is up for the task.

**Reply by honguyenminh** [reply]
I PR'ed a simple fix at
https://github.com/ppy/osu/pull/21224

**Reply by FalconTRM** [reply]
Hey, I know this is an old discussion but I prefer the controls that RBQ wanted and features for it they discussed about, than the new controls, we should be able to have an option to choose what controls we want..

**Reply by honguyenminh** [reply]
Maybe we will when we manage to add proper support for touch scheme customization.
Right now the controls (for all modes even) are hard coded, will need to make an entire new system to facilitate that.
